plugins/gatsby-jenkinsci-fieldextensions/gatsby-node.test.js

```diff
@@ -18,15 +18,18 @@ describe('gatsby-node', () => {
             });
         });
         describe('machineVersion', () => {
-            it('0.0.0', () => {
-                const value = fieldExtensions.machineVersion({field: 'version'}).resolve({version: '0.0.0'});
-                expect(value).toBe('00000_00000_00000_00000');
-            });
-            it('4.7.1.1', () => {
-                const value = fieldExtensions.machineVersion({field: 'version'}).resolve({version: '4.7.1.1'});
-                expect(value).toBe('00004_00007_00001_00001');
-            });
+            const versions = ['0.1', '1.0-b9', '1.0-b10', '1.0-rc2', '1.0', '1.0.9', '1.0.10',
```

Comment:
I think this will be confusing when it fails.

How about take the unsorted list. Map and call machine version. Then do a array compare with an expected list. Then just will show more context

It's probably fine though

Reply:
this acts as several tests, each doing one assertion. I've made the test names a bit more cllear:
```
➤ YN0000: [@jenkins-cd/gatsby-jenkinsci-fieldextensions]:         √ Version 0.1 should be less than 1.0-b9 (19 ms)
➤ YN0000: [@jenkins-cd/gatsby-jenkinsci-fieldextensions]:         √ Version 1.0-b9 should be less than 1.0-b10 (1 ms)
➤ YN0000: [@jenkins-cd/gatsby-jenkinsci-fieldextensions]:         √ Version 1.0-b10 should be less than 1.0-rc2 (1 ms)
➤ YN0000: [@jenkins-cd/gatsby-jenkinsci-fieldextensions]:         √ Version 1.0-rc2 should be less than 1.0
➤ YN0000: [@jenkins-cd/gatsby-jenkinsci-fieldextensions]:         √ Version 1.0 should be less than 1.0.9
➤ YN0000: [@jenkins-cd/gatsby-jenkinsci-fieldextensions]:         √ Version 1.0.9 should be less than 1.0.10
➤ YN0000: [@jenkins-cd/gatsby-jenkinsci-fieldextensions]:         √ Version 1.0.10 should be less than 1.0.10-2
➤ YN0000: [@jenkins-cd/gatsby-jenkinsci-fieldextensions]:         √ Version 1.0.10-2 should be less than 1.1
➤ YN0000: [@jenkins-cd/gatsby-jenkinsci-fieldextensions]:         √ Version 1.1 should be less than 2.0
➤ YN0000: [@jenkins-cd/gatsby-jenkinsci-fieldextensions]:         √ Version 2.0 should be less than 4.7.1.1 (1 ms)
```
I think failures of individual comparisons are easier to debug than failure to compare whole arrays.

Reply:
It would be better if we could do assrtions for string comparison in Jest directly, but it's not supported ATM https://github.com/jest-community/jest-extended/issues/220

Reply:
Okay cool. I'll do a real review tomorrow but it looks good so far 

I was worried it would say expected 1 to be -1 or something